### ISSUE TITLE
Make sure RefreshContainer samples will stop their timers

### ISSUE THREAD

**Issue by Youssef1313** [unoplatform/uno#15062]
### Current behavior

https://github.com/unoplatform/uno/blob/5b2d3eb6d053c1bfd7ce773b3a3a6fed2c064db7/src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RefreshContainerTests/RefreshContainerPage.xaml.cs#L78-L79

https://github.com/unoplatform/uno/blob/5b2d3eb6d053c1bfd7ce773b3a3a6fed2c064db7/src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RefreshContainerTests/RefreshContainerOnImagePage.xaml.cs#L60

It's unlikely we'll be hitting these. See https://github.com/unoplatform/uno/issues/15059#issuecomment-1891551501 for why.

We could either try to fix the fact that we don't get OnNavigatedFrom or switch to Unloaded.

### Works on UWP/WinUI

None

**Comment by MartinZikmund** [comment]
Yes, I think we should adjust the samples that are using `Frame` wrapper to properly navigate in and out.

**Comment by Youssef1313** [comment]
@MartinZikmund It's not trivial to adjust the navigation so that we get `OnNavigatedFrom`. Setting `Frame.Content` to null doesn't cause `OnNavigatedFrom` to be called.

For now, I'm workarounding that so we unsubscribe in Unloaded.

**Comment by MartinZikmund** [comment]
@Youssef1313 i was thinking to fake it a little bit by creating a Frame and pushing a blank Page there first. Then you could just navigate to a sample and back maybe 🤔

**Comment by Youssef1313** [comment]
Doing this after every sample may affect performance (for snapshot tests specifically).

Maybe try to refactor this to use a singleton frame?
